### DISCUSSION TITLE
Remove lightline support poll

### DISCUSSION THREAD

**Discussion by ellisonleao** [ellisonleao/gruvbox.nvim#268] [Polls]
Following [airline removal](https://github.com/ellisonleao/gruvbox.nvim/issues/95) that happened some time ago, i am also considering removing [lightline](https://github.com/itchyny/lightline.vim)  support as well, focusing this plugin to just support lua based plugins. I am starting a poll to see what you guys think about it. Please vote until the end of September 2023

**Comment by polyzen** [comment]
Seems like a good place to mention crystalline just recently got Lua config support:
https://github.com/rbong/vim-crystalline/blob/master/examples/neovim_lua_examples.md

It includes its own gruvbox light/dark themes.

**Comment by ellisonleao** [comment]
Removed in 31cc564

**Comment by perrin4869** [comment]
hm... I was actually still using `lightline` with `gruvbox.nvim`, and this broke my config 😅
https://github.com/perrin4869/dotfiles/pull/2691
Is there a way to migrate the lightline themes? I tried to place them in `~/.local/share/nvim/autoload/lightline/colorscheme/gruvbox.nvim`, but haven't had success thus far

```vimscript
let s:theme = v:lua.require("gruvbox.groups").setup()

let s:bg0 = s:theme.GruvboxBg0.fg
let s:bg1 = s:theme.GruvboxBg1.fg
let s:bg2 = s:theme.GruvboxBg2.fg
let s:bg4 = s:theme.GruvboxBg4.fg
let s:fg1 = s:theme.GruvboxFg1.fg
let s:fg4 = s:theme.GruvboxFg4.fg

let s:yellow = s:theme.GruvboxYellow.fg
let s:blue = s:theme.GruvboxBlue.fg
let s:aqua = s:theme.GruvboxAqua.fg
let s:orange = s:theme.GruvboxOrange.fg
let s:red = s:theme.GruvboxRed.fg
let s:green = s:theme.GruvboxGreen.fg

let s:palette = {
    \ 'normal': {
    \   'left': [[s:bg0, s:fg4, 'bold'], [s:fg4, s:bg2]],
    \   'middle': [[s:fg4, s:bg1]],
    \   'right': [[s:bg0, s:fg4], [s:fg4, s:bg2]],
    \   'error': [[s:bg0, s:red]],
    \   'warning': [[s:bg0, s:yellow]]
    \ },
    \ 'insert': {
    \   'left': [[s:bg0, s:blue, 'bold'], [s:fg1, s:bg2]],
    \   'middle': [[s:fg4, s:bg1]],
    \   'right': [[s:bg0, s:blue], [s:fg1, s:bg2]]
    \ },
    \ 'inactive': {
    \   'left': [[s:bg4, s:bg1]],
    \   'middle': [[s:bg4, s:bg1]],
    \   'right': [[s:bg4, s:bg1], [s:bg4, s:bg1]]
    \ },
    \ 'terminal': {
    \   'left': [[s:bg0, s:green, 'bold'], [s:fg1, s:bg2]],
    \   'middle': [[s:fg4, s:bg1]],
    \   'right': [[s:bg0, s:green], [s:fg1, s:bg2]]
    \ },
    \ 'replace': {
    \   'left': [[s:bg0, s:aqua, 'bold'], [s:fg1, s:bg2]],
    \   'middle': [[s:fg4, s:bg1]],
    \   'right': [[s:bg0, s:aqua], [s:fg1, s:bg2]]
    \ },
    \ 'visual': {
    \   'left': [[s:bg0, s:orange, 'bold'], [s:bg0, s:bg4]],
    \   'middle': [[s:fg4, s:bg1]],
    \   'right': [[s:bg0, s:orange], [s:bg0, s:bg4]]
    \ },
    \ 'tabline': {
    \   'left': [[s:fg4, s:bg2]],
    \   'middle': [[s:bg0, s:bg4]],
    \   'right': [[s:bg0, s:orange]],
    \   'tabsel': [[s:bg0, s:fg4]]
    \ }
  \ }

let g:lightline#colorscheme#gruvbox#palette = lightline#colorscheme#fill(s:palette)
```

**Reply by perrin4869** [reply]
ok, managed to get it working using:

```vimscript
let s:theme = v:lua.require("gruvbox.groups").setup(v:lua.require("gruvbox").config)
```

**Reply by BarrensZeppelin** [reply]
Thanks!

**Reply by ellisonleao** [reply]
@perrin4869 any chance you could also move to another statusline plugin, using lua? it's way more reliable IMHO

**Reply by perrin4869** [reply]
Hm... Don't plan to migrate soon honestly, it works just fine I think...
What kind of problems did you have with it?

**Reply by ellisonleao** [reply]
@perrin4869 basically the vimscript support which honestly it's already a no-go for me in the neovim ecosystem

**Reply by perrin4869** [reply]
I see, personally I'm totally fine moving the `lightline` integration into my repo, no need for `gruvbox.nvim` to maintain it

I did briefly check `lualine` and `crystalline`, but I didn't find that they would improve my setup, so opted to pass on them for now. They aren't much more performant than `lightline` according to their benchmarks. However, lightline does seem to be somewhat unmaintained so I might migrate in the future

**Reply by perrin4869** [reply]
In case anyone is interested in the integration to work on the latest HEAD release, you can replace with:

```vimscript
function! ReturnHighlightTerm(group, term)
   " Store output of group to variable
   let output = execute('hi ' . a:group)

   " Find the term we're looking for
   return matchstr(output, a:term.'=\zs\S*')
endfunction

let s:bg0 = ReturnHighlightTerm('GruvboxBg0', 'guifg')
let s:bg1 = ReturnHighlightTerm('GruvboxBg1', 'guifg')
let s:bg2 = ReturnHighlightTerm('GruvboxBg2', 'guifg')
let s:bg4 = ReturnHighlightTerm('GruvboxBg4', 'guifg')
let s:fg1 = ReturnHighlightTerm('GruvboxFg1', 'guifg')
let s:fg4 = ReturnHighlightTerm('GruvboxFg4', 'guifg')

let s:yellow = ReturnHighlightTerm('GruvboxYellow', 'guifg')
let s:blue = ReturnHighlightTerm('GruvboxBlue', 'guifg')
let s:aqua = ReturnHighlightTerm('GruvboxAqua', 'guifg')
let s:orange = ReturnHighlightTerm('GruvboxOrange', 'guifg')
let s:red = ReturnHighlightTerm('GruvboxRed', 'guifg')
let s:green = ReturnHighlightTerm('GruvboxGreen', 'guifg')

```

**Reply by adiSuper94** [reply]
Also lightline still seems to be faster (on my machines), so want to stick to it.
Weird that I have been gruvbox + lightline for over year now and never faced any issues.
It has broken lightline color theme only in nvim 0.11.
